addons/sale_loyalty/models/sale_order.py

```diff
@@ -121,9 +123,33 @@ def _add_loyalty_history_lines(self):
                 'card_id': coupon.id,
                 'used': cost,
                 'issued': issued,
+                'available_issued_points': issued,
             })
+            if cost:
+                # for redemption of loyalty points used as reward
+                reward_records.append({
+                    'card_id': coupon.id,
+                    'points_to_redeem': cost,
+                })
+            if issued and coupon.points < 0:
+                # for compensating the negative card balance to cover debt
+                compensation_records.append({
+                    'card_id': coupon.id,
+                    'points_to_redeem': abs(coupon.points),
```

Comment:
Why are we adding these? If I understand correctly, it's never used later. We search in compensate_existing_debts for lines with negative points, so there is no point in passing this parameter since we pass card id.